desc/coils.py

```diff
@@ -3050,3 +3050,64 @@ def initialize_saddle_coils(eq, num_coils, r_over_a=0.5, offset=2.0, position="o
 
     windowpane_coilset = CoilSet(windowpane_coils, NFP=int(eq.NFP), sym=eq.sym)
     return windowpane_coilset
+
+
+def initialize_helical_coils(eq, num_coils, r_over_a=2.0, helicity=(1, 1), npts=100):
+    """Initialize a CoilSet of helical coils for stage 2 optimization.
+
+    The coils will be roughly a constant distance from the plasma surface as they wind
+    around. The currents will be set to match the equilibrium required poloidal
+    linking current.
+
+    The coils will be ``SplineXYZCoil``, if another type is desired use
+    ``coilset.to_FourierXYZ(N=...)``, etc.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Stage 1 equilibrium the coils are being optimized for.
+    num_coils : int
+        Number of coils to create.
+    r_over_a : float
+        Approximate minor radius of the coils, in units of equilibrium minor radius. The
+        actual coils will be contoured to keep a roughly constant offset from the plasma
+        surface.
+    helicity : tuple of int
+        (M,N) - How many times each coil should link the plasma poloidally and
+        toroidally. Note that M is the poloidal linking number per field period, so the
+        total linking number will be M*NFP.
+    npts : int
+        How many points to use when creating the coils. Equilibria with very high NFP
+        may need more points.
+
+    Returns
+    -------
+    coilset : CoilSet of FourierPlanarCoil
+        Planar coils centered on magnetic axis, with appropriate symmetry.
+    """
+    a = eq.compute("a")["a"]
+    G = eq.compute("G", grid=LinearGrid(rho=1.0))["G"]
+    M = helicity[0] * eq.NFP
+    N = helicity[1]
+
+    s = np.linspace(0, 2 * np.pi, npts, endpoint=False)
+
+    theta = M * s
+    zeta = N * s
+
+    theta %= 2 * np.pi
+    zeta %= 2 * np.pi
+
+    theta_offset = np.linspace(0, 2 * np.pi, num_coils, endpoint=False)
```

Comment:
this will always have a coil near theta=zeta=0, is that something we need an option for?

Reply:
I think it is fine, they could always rotate the coil around if they don't want it there

Reply:
our symmetry stuff does not reduce effort for helical coils anyways right now iirc